core/ledger/kvledger/txmgmt/statedb/stateleveldb/stateleveldb.go

```diff
@@ -21,10 +21,12 @@ import (
 var logger = flogging.MustGetLogger("stateleveldb")
 
 var (
-	dataKeyPrefix    = []byte{'d'}
-	nsKeySep         = []byte{0x00}
-	lastKeyIndicator = byte(0x01)
-	savePointKey     = []byte{'s'}
+	dataKeyPrefix               = []byte{'d'}
+	dataKeyStopper              = []byte{'e'}
+	nsKeySep                    = []byte{0x00}
+	lastKeyIndicator            = byte(0x01)
+	savePointKey                = []byte{'s'}
+	fullScanIteratorValueFormat = byte(1)
```

Comment:
Yes, we can avoid this if return VersionedValue instead of bytes. But since, here we are dealing with potentially large data, avoiding marshalling/unmarshalling and memory allocations, as much as possible should be desired.